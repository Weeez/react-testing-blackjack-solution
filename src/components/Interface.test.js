import React from 'react'
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Interface from './Interface';
import Outcome from './Outcome'

configure({adapter: new Adapter()});

describe('<Interface />', () => {

    let wrapper;

    beforeEach(() => {
        wrapper = shallow(
            <Interface />
        )
    })

    it('has an Outcome child component', () => {
        expect(wrapper.find(Outcome)).toHaveLength(1);
    })

    it('has two children with .score class name', () => {
        expect(wrapper.find('.score')).toHaveLength(2);
    })

    it('receives scores correctly', () => {
        wrapper = shallow(<Interface computerScore = {17} />);
        expect(wrapper.find('.score').first().text()).toEqual('Dealer Score : 17');
    })

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })

})