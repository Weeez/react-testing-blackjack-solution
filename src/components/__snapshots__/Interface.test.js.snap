// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Interface /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Interface />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Outcome
          message={undefined}
/>,
        <div>
          <a
                    className="score"
          >
                    Dealer Score : 
          </a>
          <a
                    className="score"
          >
                    Player Score : 
          </a>
</div>,
        <button
          id="deal-button"
          onClick={undefined}
>
          Deal
</button>,
        <button
          id="hit-button"
          onClick={undefined}
>
          Hit
</button>,
        <button
          id="stand-button"
          onClick={undefined}
>
          Stand
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "message": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <a
              className="score"
>
              Dealer Score : 
</a>,
            <a
              className="score"
>
              Player Score : 
</a>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Dealer Score : ",
                undefined,
              ],
              "className": "score",
            },
            "ref": null,
            "rendered": Array [
              "Dealer Score : ",
              undefined,
            ],
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Player Score : ",
                undefined,
              ],
              "className": "score",
            },
            "ref": null,
            "rendered": Array [
              "Player Score : ",
              undefined,
            ],
            "type": "a",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Deal",
          "id": "deal-button",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "Deal",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Hit",
          "id": "hit-button",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "Hit",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Stand",
          "id": "stand-button",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "Stand",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Outcome
            message={undefined}
/>,
          <div>
            <a
                        className="score"
            >
                        Dealer Score : 
            </a>
            <a
                        className="score"
            >
                        Player Score : 
            </a>
</div>,
          <button
            id="deal-button"
            onClick={undefined}
>
            Deal
</button>,
          <button
            id="hit-button"
            onClick={undefined}
>
            Hit
</button>,
          <button
            id="stand-button"
            onClick={undefined}
>
            Stand
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "message": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <a
                className="score"
>
                Dealer Score : 
</a>,
              <a
                className="score"
>
                Player Score : 
</a>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Dealer Score : ",
                  undefined,
                ],
                "className": "score",
              },
              "ref": null,
              "rendered": Array [
                "Dealer Score : ",
                undefined,
              ],
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Player Score : ",
                  undefined,
                ],
                "className": "score",
              },
              "ref": null,
              "rendered": Array [
                "Player Score : ",
                undefined,
              ],
              "type": "a",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Deal",
            "id": "deal-button",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Deal",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Hit",
            "id": "hit-button",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Hit",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Stand",
            "id": "stand-button",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Stand",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
